src/types/Typechecker/TypeError.hs

```diff
@@ -799,9 +821,17 @@ instance Show Error where
     show (ActiveTraitError trait) =
         printf "Trait '%s' can only be included by active classes"
                (show trait)
-    show (UnsafeTypeArgumentError ty) =
-        printf "Cannot use non-aliasable type '%s' as type argument."
-               (show ty)
+    show (UnsafeTypeArgumentError formal ty) =
+        if isModeless ty then
+          printf "Cannot use non-aliasable type '%s' as type argument" --TODO: This seems imprecise...
```

Comment:
The `TODO` there is relevant... I'll look into when this case is hit exactly and change it accordingly. It might be that this is actually dead code.